### PR TITLE
Draft: Security: Require confirmation to start TabPy without authentication

### PR DESCRIPTION
**Pull request by jakeichikawasalesforce**
Require confirmation to continue when starting TabPy without authentication, with a warning that this is an insecure state and not recommended. The message will be:
```
WARNING: This TabPy server is not currently configured for username/password authentication. This means that, because the TABPY_EVALUATE_ENABLE feature is enabled, there is the potential that unauthenticated individuals may be able to remotely execute code on this machine. We strongly advise against proceeding without authentication as it poses a significant security risk.

Do you wish to proceed without authentication? (Y/N):
```

Bumping version to 2.9.0.

Also removing Python 3.7 from GitHub actions due to https://github.com/actions/setup-python/issues/682. Will re-add later.

### CODE REVIEW

File: tests/integration/integ_test_base.py [hunk 1]
```diff
@@ -236,11 +236,18 @@ def setUp(self):
...
 
+            # check if there is a confirmation prompt for no authentication
+            time.sleep(3)
+            with open(self.tmp_dir + "/output.txt", "r") as infile:
+                if "WARNING: No username/password" in infile.read():
+                    self.process.stdin.write(b"Y\n")
+                    self.process.stdin.flush()
+
             # give the app some time to start up...
-            time.sleep(5)
```
[the remark this answers is not in the document]
Comment (jakeichikawasalesforce) [reply tests/integration/integ_test_base.py:243]: Thanks John

File: tabpy/tabpy_server/app/app.py [hunk 2]
```diff
@@ -471,6 +469,25 @@ def _parse_pwd_file(self):
...
+        
+        if self.settings[SettingsParameters.EvaluateEnabled]:
+            confirm_no_auth_msg += ("Since TABPY_EVALUATE_ENABLE is on, unauthenticated users can execute "
+                "remote code on this machine, posing significant security risks. ")
+        
+        confirm_no_auth_msg += ("Proceeding in this insecure state is strongly discouraged.\n\n"
+            "Are you sure you want to continue without authentication? (Y/N): ")
+        confirm_no_auth_input = input(confirm_no_auth_msg)
+        
+        if confirm_no_auth_input.lower() == 'y':
```
[the remark this answers is not in the document]
Comment (jakeichikawasalesforce) [reply tabpy/tabpy_server/app/app.py]: Thanks John

File: tabpy/tabpy_server/app/app.py [hunk 3]
```diff
@@ -471,6 +469,25 @@ def _parse_pwd_file(self):
...
 
+    def _handle_configuration_without_authentication(self):
+        confirm_no_auth_msg = "\nWARNING: No username/password authentication is configured for this TabPy server. "
+        
+        if self.settings[SettingsParameters.EvaluateEnabled]:
+            confirm_no_auth_msg += ("Since TABPY_EVALUATE_ENABLE is on, unauthenticated users can execute "
+                "remote code on this machine, posing significant security risks. ")
+        
+        confirm_no_auth_msg += ("Proceeding in this insecure state is strongly discouraged.\n\n"
+            "Are you sure you want to continue without authentication? (Y/N): ")
```
[the remark this answers is not in the document]
Comment (jakeichikawasalesforce) [reply tabpy/tabpy_server/app/app.py]: Thanks John.

**Review by johng42 — APPROVED** [review]
LGTM

**Comment by jakeichikawasalesforce** [comment]
@johng42 thanks for the review! I'm recreating a clean branch and PR because this issue (https://github.com/actions/setup-python/issues/682) threw me for a loop and I have a bunch of unnecessary commits in this branch. Wanted to clean that up a bit.

**Pull request closed without merging**